### PR TITLE
Enabling Workload Identity and GCSFuse driver flags added.

### PR DESCRIPTION
**Pull request by sharabiani**
## Fixes / Features
- `--enable-workload-identity` flag implemented for `cluster create' command. It enables Workload Identity Federation on the cluster and node-pools.
- `--enable-gcsfuse-csi-driver` flag implemented for `cluster create' command. It enables GSCFuse driver on the cluster. This also enables Workload Identity Federation.'

## Testing / Documentation
Manual tests.

- [ y/n ] Tests pass
- [ y/n ] Appropriate changes to documentation are included in the PR

### CODE REVIEW

File: src/xpk/commands/cluster.py
Comment (Obliviour) [inline src/xpk/commands/cluster.py:93]: Nit: curious if you want to check for the case `if not args.enable_workload_identity and args.enable_gcsfuse_driver: ... return error`

File: src/xpk/parser/cluster.py
Comment (Obliviour) [inline src/xpk/parser/cluster.py:453]: One discussion we can think through is if we want to support workload-identity separately from gcsfuse. Do we see a need to support workload-identity by itself such as the additional dev support of the feature by itself in XPK? The alternative is we just support the current request (from what I am aware) is just gcsfuse-csi-driver support. If we don't currently see value for `--enable-workload-identity` flag by itself, we don't need to add it now (one way door)! Curious what you think!

File: src/xpk/core/core.py
Comment (Obliviour) [inline src/xpk/core/core.py:1620]: nit to add comment saying the xpk doesn't support disabling WI on clusters that have it enabled already? (Which makes sense, not saying we should add this)

**Review by Obliviour — APPROVED** [review]
LGTM, great changes! I have some questions about --enable-workload-identity support by itself and some nits!

File: src/xpk/commands/cluster.py
Comment (sharabiani) [reply src/xpk/commands/cluster.py]: does false value for args.enable_workload_identity means user explicitly set it to false? or it can also mean user did not provided the flag?
I assumed the latter is the case, and we'll automatically enable WI if gsfuse is enabled.
What do you think?

File: src/xpk/parser/cluster.py
Comment (sharabiani) [reply src/xpk/parser/cluster.py:453]: That's what I'm waiting for the decision. For now it's possible to only provide only --enable-gcsfuse-csi-driver flag which will enable WI, too.
We can remove the WI flag later if we no need for that. What do you think?

File: src/xpk/core/core.py
Comment (sharabiani) [reply src/xpk/core/core.py:1618]: done

**Comment by PBundyra** [comment]
LGTM

File: src/xpk/parser/cluster.py
Comment (Obliviour) [reply src/xpk/parser/cluster.py:453]: > We can remove the WI flag later if we no need for that. What do you think?

Yes I agree we can remove it, though I would prefer to remove it now since it is harder to remove later! But I'll leave the decision up to you here :)

Comment (sharabiani) [reply src/xpk/parser/cluster.py:453]: I'll leave it to @PBundyra as we discussed and close this PR.

**Pull request merged**
